### PR TITLE
Update GitHub Actions

### PR DESCRIPTION
**Pull request by rettinghaus**
#### Brief description of what is fixed or changed
This PR just updates the used GitHub Actions to latest versions.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscargus** [comment]
Thanks!

```
This author is not included in the .mailmap file:
Klaus Rettinghaus <klaus.rettinghaus@enote.com>

The .mailmap file needs to be updated because there are commits with
unrecognised author/email metadata.

For instructions on updating the .mailmap file see:
https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file
```

Apart from that all look good!

File: .github/workflows/comment-on-pr.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         # here to be able to comment on the PR
 
       - name: Download benchmarks
-        uses: dawidd6/action-download-artifact@v2
+        uses: dawidd6/action-download-artifact@v3
```
Comment (oscarbenjamin) [inline .github/workflows/comment-on-pr.yml:28]: I don't think that this should have been changed. It seems there is no v3:
https://github.com/sympy/sympy/actions/runs/3328189773/jobs/5503897421
https://github.com/dawidd6/action-download-artifact

Comment (oscarbenjamin) [reply .github/workflows/comment-on-pr.yml:28]: I guess that's why there is no benchmarks report on this PR.

Comment (rettinghaus) [reply .github/workflows/comment-on-pr.yml:28]: Oh my. Sorry for messing this up. I'll open another PR with a fix.

Comment (oscargus) [reply .github/workflows/comment-on-pr.yml:28]: Sorry. My bad for not checking better.

Comment (rettinghaus) [reply .github/workflows/comment-on-pr.yml:28]: See #24181

**Pull request merged**